plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/JdbcRecordSetProvider.java

```diff
@@ -48,14 +48,19 @@ public JdbcRecordSetProvider(JdbcClient jdbcClient, @ForRecordCursor ExecutorSer
     public RecordSet getRecordSet(ConnectorTransactionHandle transaction, ConnectorSession session, ConnectorSplit split, ConnectorTableHandle table, List<? extends ColumnHandle> columns)
     {
         JdbcSplit jdbcSplit = (JdbcSplit) split;
-        JdbcTableHandle jdbcTable = (JdbcTableHandle) table;
 
-        // In the current API, the columns (and order) needed by the engine are provided via an argument to this method. Make sure we can
-        // satisfy the requirements using columns which were recorded in the table handle.
-        // If no columns are recorded, it means that applyProjection never got called (e.g., in the case all columns are being used) and all
-        // table columns should be returned. TODO: this is something that should be addressed once the getRecordSet API is revamped
-        jdbcTable.getColumns()
-                .ifPresent(tableColumns -> verify(ImmutableSet.copyOf(tableColumns).containsAll(columns)));
+        ConnectorTableHandle tableHandle = table;
+
+        if (table instanceof JdbcTableHandle jdbcTable) {
+            // In the current API, the columns (and order) needed by the engine are provided via an argument to this method. Make sure we can
+            // satisfy the requirements using columns which were recorded in the table handle.
+            // If no columns are recorded, it means that applyProjection never got called (e.g., in the case all columns are being used) and all
+            // table columns should be returned. TODO: this is something that should be addressed once the getRecordSet API is revamped
+            jdbcTable.getColumns()
+                    .ifPresent(tableColumns -> verify(ImmutableSet.copyOf(tableColumns).containsAll(columns)));
```

Comment:
Why not for procedure handle?
Have same check for them too

Reply:
In case of `ProcedureHandle` we won't face the `no columns` kind of a situation - plus the columns are populated when we are creating the table handle. Add the `ProcedureHandle` doesn't support project pushdown there is a good chance the column we pass on `RecordSetProvider` might be a subset of overall columns present in `ProcedureHandle`.